app/controllers/api/v0/api_controller.rb

```diff
@@ -156,7 +156,11 @@ def omni_search
   end
 
   def delegates
-    paginate json: UserGroup.delegate_regions.flat_map(&:active_users)
+    delegate_users = UserGroup.roles_of_group_type(
```

Comment:
I am not really a fan of providing one "god method" that does selecting, filtering and sorting all in one.
In this particular example, you are not applying any filter params at all, but the `roles_of_group_type` method will still run through the `filter` method needlessly.
This is especially problematic because your current implementation of the `filter_roles` method loops through all roles objects in-memory, so in this specific case for Delegates, the roles are being loaded and looped through (in the `reject` part of the `UserRole.filter_roles` implementation) for literally no reason.
It would be _much_ more efficient to just return the `UserRole.includes(...).where(...)` part directly, especially for a paginated endpoint like this.

Reply:
For now, I've reverted the changes in this file. I'll come back to it in another PR as I've some plans for this 😬